cymbaline/generators.py

```diff
@@ -0,0 +1,63 @@
+from cymbaline.data import DataProvider
+from cymbaline.language import LanguageToolkit
+
+import random
+
+
+class Generator:
+    def __init__(self, context_word=None, context_vector=None):
+    """Initiates Generator with either single word or vector, which defines the context"""
+        if(context_word == None and context_vector == None):
+            raise TypeError("Either `context_word` or `context_vector` argument is required.")
+
+        self._context_word = context_word
+        self._context_vector = context_vector
+
+        self.lt = LanguageToolkit(context_word=context_word, context_vector=context_vector)
+    
+    @property
+    def context_vector(self):
+        return self.context_vector
+
+    @context_vector.setter
+    def context_vector(self, value):
+        self.lt = LanguageToolkit(context_word=None, context_vector=value)
+
+    @property
+    def context_word(self):
+        return self.context_vector
+    
+    @context_word.setter
+    def context_word(self, value):
+        self.lt = LanguageToolkit(context_word=value, context_vector=None)
+
+    def generate(self):
+        raise NotImplementedError("Abstract `Generator` base class method has been called.")
+
+    def _generate_n_syllables_stich(self, n):
+        stich = ""
+
+        while(n > 0):
+            word_syllables_count = random.randint(1, n)
+            if(stich != ""):
```

Comment:
Python makes reasonable choices when it comes to conditional statements. An empty `str`, `list`, `dict` or other structures will automatically evaluate to `False`.
```suggestion
            if stich:
```

Reply:
Your proposal isn't straightforward for any person without good Python command.

Reply:
Python tries to replicate writing in natural language as much as possible. It is not a coincidence, that most of the available structures are meant to be easily understandable, eg.
```python
for element in container:
```
And so the default approach for conditional statements in Python is to use the defined behavior of boolean type casting, which can be set of every class.
```python
if statement:
```
Instead of:
```python
if statement is True:
```

This way of specifying conditions is explained in most Python tutorials for beginners - in fact, that probably would be included in the first lesson. 

There is also another issue with your approach, one that could be more serious than just aesthetics. It is not just the syntax we are after. 
Unlike C/C++, Python is a language that uses duck typing. You can't be expecting, that the `stich` variable will always be a `string` - sometimes, you may encounter non trivial erors. We can assume, that for some reason the value of `stich`'s is currently set `None` (which happens frequently in Python) - in this case, your conditional statement only checks for empty strings, thus allowing the block of code to be executed with trivially incorrect value (which will almost definitely result in the termination of the program).

Reply:
It is still not straightforward, because one has to know, an empty string evaluates to `False`. Anyway, I introduced this change before your last comment was published. Now I am way more convinced, because of `None` check though.